test/parallel/test-timers-promisified.js

```diff
@@ -392,18 +392,23 @@ process.on('multipleResolves', common.mustNotCall());
   // Check that the timing is correct
   let pre = false;
   let post = false;
-  setPromiseTimeout(1).then(() => pre = true);
-  const iterable = setInterval(() => {}, 2);
-  const iterator = iterable[Symbol.asyncIterator]();
-
-  iterator.next().then(common.mustCall(() => {
-    assert.ok(pre, 'interval ran too early');
-    assert.ok(!post, 'interval ran too late');
-    return iterator.next();
-  })).then(common.mustCall(() => {
-    assert.ok(post, 'second interval ran too early');
-    return iterator.return();
-  }));
 
-  setPromiseTimeout(3).then(() => post = true);
+  Promise.all([
+    setPromiseTimeout(1).then(() => pre = true),
+    new Promise((res) => {
+      const iterable = timerPromises.setInterval(200);
+      const iterator = iterable[Symbol.asyncIterator]();
+
+      iterator.next().then(common.mustCall(() => {
+        assert.ok(pre, 'interval ran too early');
+        assert.ok(!post, 'interval ran too late');
+        return iterator.next();
+      })).then(common.mustCall(() => {
+        assert.ok(post, 'second interval ran too early');
+        res();
+        return iterator.return();
+      }));
```

Comment:
Shouldn't you resolve the promise at the end?
```suggestion
        return iterator.return();
      }).then(res);
```